siliconcompiler/schema/schema_obj.py

```diff
@@ -187,7 +187,7 @@ def __get(self, *keypath, field='value', job=None, step=None, index=None):
             raise ValueError(f'Invalid field {field}')
 
     ###########################################################################
-    def set(self, *args, field='value', clobber=True, step=None, index=None):
+    def set(self, *args, field='value', clobber=True, step=None, index=None, dependency=''):
```

Comment:
it might be nice if the schema didn't have this and instead relied on Chip class (there is some awkward integration here that requires the same edits in several places).

Reply:
Yes but then I need to inspect the `cfg['type']` of the keypath in `Chip`.
If you think that would be prettier I can do that too.

Reply:
lets leave this for now